main_logic/core.py

```
@@ -2086,6 +2087,63 @@ async def trigger_agent_callbacks(self) -> None:
+        _lang = normalize_language_code(self.user_language, format='short')
+        from config.prompts_proactive import get_greeting_prompt
+        from memory_server import _format_elapsed
+        template = get_greeting_prompt(gap_seconds, _lang)
+        if not template:
+            return
+
+        elapsed = _format_elapsed(_lang, gap_seconds)
+        instruction = template.format(elapsed=elapsed, name=self.lanlan_name, master=self.master_name)
+        logger.info("[%s] trigger_greeting: gap=%.0fs elapsed=%s, delivering", self.lanlan_name, gap_seconds, elapsed)
+
+        # 如果已有 text session 且空闲，直接投递
```

Ruff (0.15.7)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)